source/core_http_client.c

```diff
@@ -2116,9 +2115,20 @@ static int findHeaderValueParserCallback( http_parser * pHttpParser,
                     "RequestedField=%.*s, ValueLocation=0x%p",
                     ( int ) ( pContext->fieldLen ), pContext->pField, pValueLoc ) );
 
-        /* Populate the output parameters with the location of the header value in the response buffer. */
-        *pContext->pValueLoc = pValueLoc;
-        *pContext->pValueLen = valueLen;
+        if( valueLen > 0 )
+        {
+            /* Populate the output parameters with the location of the header
+             * value in the response buffer. */
+            *pContext->pValueLoc = pValueLoc;
+            *pContext->pValueLen = valueLen;
+        }
+        else
+        {
+            /* It is not invalid according to rfc2616 to have an empty header
+             * value. */
+            *pContext->pValueLoc = NULL;
```

Comment:
Do we want to not provide the location of the empty header value in the response buffer to the application?

Reply:
http-parser passes in the next start of the header field to the const char* location parameter of this callback. It seems more clean to just give the customers a NULL, in this case. There could be no spaces between the header field separator and the header line endings.

Reply:
Okay, as the `http-parser` is providing the pointer to the next header field, then it makes sense to provide `NULL`